static/js/controllers/reports.js

```diff
@@ -131,34 +187,60 @@ var _ = require('underscore'),
           return DB.get()
             .get(report)
             .then(FormatDataRecord);
-        } else {
-          return FormatDataRecord(report);
         }
+        return FormatDataRecord(report);
       };
 
       $scope.refreshReportSilently = function(report) {
-        _fetchFormattedReport(report)
+        return _fetchFormattedReport(report)
           .then(function(doc) {
-              _setSelected(doc[0]);
-            })
+            _setSelected(doc[0]);
+          })
           .catch(function(err) {
             $log.error('Error fetching formatted report', err);
           });
       };
 
+      $scope.deselectReport = function(report) {
+        for (var i = 0; i < $scope.selected.length; i++) {
```

Comment:
whoa iterating instead of underscore, don't remember what it is any more, hurts brain, can't read